<a href="https://colab.research.google.com/github/Alan-O-R/CP1_Semestre2/blob/main/Renovaveis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# BAIXAR A BIBLIOTECA

In [ ]:
import pandas as pd

# NOMES:
- Joao Ribeiro Rm: 570562
- Alan Otalvaro Rm: 571794
- Gabriel de Paula RM: 573195
- Joao Pero Envangelista Rm: 573899
- Enzo ribeiro Rm: 569429
- João Lucas silva Rm: 573875


# Primeiro DataSet
## Fonte UCI : Appliances Energy Prediction
### Consumo de eletrodomésticos em uma residência, associado a temperatura, umidade e condições ambientais.

In [ ]:
energydata = pd.read_csv('energydata_complete.csv')
Rn_energydata = energydata.rename(columns={
    "Appliances" : "Consumo_Eletrodomesticos",
    "T_out": "temp_ext",
    "RH_out": "umid_ext",
    "Windspeed":"vento",
})

In [ ]:
Rn_energydata.head(5)
Rn_energydata.shape[0]
Rn_energydata.info()
Rn_energydata.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19735 entries, 0 to 19734
Data columns (total 29 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   date                      19735 non-null  object 
 1   Consumo_Eletrodomesticos  19735 non-null  int64  
 2   lights                    19735 non-null  int64  
 3   T1                        19735 non-null  float64
 4   RH_1                      19735 non-null  float64
 5   T2                        19735 non-null  float64
 6   RH_2                      19735 non-null  float64
 7   T3                        19735 non-null  float64
 8   RH_3                      19735 non-null  float64
 9   T4                        19735 non-null  float64
 10  RH_4                      19735 non-null  float64
 11  T5                        19735 non-null  float64
 12  RH_5                      19735 non-null  float64
 13  T6                        19735 non-null  float64
 14  RH_6  

,Consumo_Eletrodomesticos,lights,T1,RH_1,T2,RH_2,T3,RH_3,T4,RH_4,...,T9,RH_9,temp_ext,Press_mm_hg,umid_ext,vento,Visibility,Tdewpoint,rv1,rv2
count,19735.000000,19735.000000,19735.000000,19735.000000,19735.000000,19735.000000,19735.000000,19735.000000,19735.000000,19735.000000,...,19735.000000,19735.000000,19735.000000,19735.000000,19735.000000,19735.000000,19735.000000,19735.000000,19735.000000,19735.000000
mean,97.694958,3.801875,21.686571,40.259739,20.341219,40.420420,22.267611,39.242500,20.855335,39.026904,...,19.485828,41.552401,7.411665,755.522602,79.750418,4.039752,38.330834,3.760707,24.988033,24.988033
std,102.524891,7.935988,1.606066,3.979299,2.192974,4.069813,2.006111,3.254576,2.042884,4.341321,...,2.014712,4.151497,5.317409,7.399441,14.901088,2.451221,11.794719,4.194648,14.496634,14.496634
min,10.000000,0.000000,16.790000,27.023333,16.100000,20.463333,17.200000,28.766667,15.100000,27.660000,...,14.890000,29.166667,-5.000000,729.300000,24.000000,0.000000,1.000000,-6.600000,0.005322,0.005322
25%,50.000000,0.000000,20.760000,37.333333,18.790000,37.900000,20.790000,36.900000,19.530000,35.530000,...,18.000000,38.500000,3.666667,750.933333,70.333333,2.000000,29.000000,0.900000,12.497889,12.497889
50%,60.000000,0.000000,21.600000,39.656667,20.000000,40.500000,22.100000,38.530000,20.666667,38.400000,...,19.390000,40.900000,6.916667,756.100000,83.666667,3.666667,40.000000,3.433333,24.897653,24.897653
75%,100.000000,0.000000,22.600000,43.066667,21.500000,43.260000,23.290000,41.760000,22.100000,42.156667,...,20.600000,44.338095,10.408333,760.933333,91.666667,5.500000,40.000000,6.566667,37.583769,37.583769
max,1080.000000,70.000000,26.260000,63.360000,29.856667,56.026667,29.236000,50.163333,26.200000,51.090000,...,24.500000,53.326667,26.100000,772.300000,100.000000,14.000000,66.000000,15.500000,49.996530,49.996530


In [ ]:
consumo_max = Rn_energydata["Consumo_Eletrodomesticos"].max()
print(f"Maior consumo de eletrodomésticos: {consumo_max}")

Maior consumo de eletrodomésticos: 1080


In [ ]:
limiar_consumo = 0.7 * consumo_max
df_consumo_alto = Rn_energydata[Rn_energydata["Consumo_Eletrodomesticos"] > limiar_consumo]

print(f"Limiar (70% do máximo): {limiar_consumo}")
df_consumo_alto.head()

Limiar (70% do máximo): 756.0


,date,Consumo_Eletrodomesticos,lights,T1,RH_1,T2,RH_2,T3,RH_3,T4,...,T9,RH_9,temp_ext,Press_mm_hg,umid_ext,vento,Visibility,Tdewpoint,rv1,rv2
432,2016-01-14 17:00:00,910,0,21.463333,41.693333,20.856667,38.363333,21.666667,43.930000,18.700000,...,17.00,43.466667,4.600000,747.900000,76.000000,7.000000,40.0,0.700000,18.937972,18.937972
443,2016-01-14 18:50:00,790,10,22.166667,40.500000,21.700000,38.626667,22.166667,42.133333,18.790000,...,16.89,42.326667,4.083333,748.433333,76.833333,6.833333,40.0,0.350000,30.487447,30.487447
731,2016-01-16 18:50:00,1080,30,21.930000,42.766667,21.040000,38.080000,20.700000,40.633333,20.890000,...,17.79,40.672500,2.083333,765.033333,85.500000,3.666667,40.0,-0.083333,27.408521,27.408521
867,2016-01-17 17:30:00,800,10,21.500000,36.760000,20.100000,36.077273,20.790000,37.260000,21.000000,...,18.10,37.260000,0.000000,764.850000,78.000000,2.000000,40.0,-3.500000,31.552437,31.552437
868,2016-01-17 17:40:00,790,10,21.566667,38.626667,20.100000,37.040909,20.790000,37.260000,21.066667,...,18.10,37.290000,-0.266667,764.833333,79.666667,2.000000,40.0,-3.466667,36.170216,36.170216


In [ ]:
n_selecionados = df_consumo_alto.shape[0]
n_total = Rn_energydata.shape[0]
percentual = (n_selecionados / n_total) * 100

print(f"Registros acima do limiar: {n_selecionados}")
print(f"Total de registros na amostra: {n_total}")
print(f"Percentual da amostra: {percentual:.2f}%")

Registros acima do limiar: 31
Total de registros na amostra: 19735
Percentual da amostra: 0.16%


In [ ]:
temp_media_T1 = Rn_energydata["T1"].mean()

df_consumo_e_temp = Rn_energydata[
    (Rn_energydata["Consumo_Eletrodomesticos"] > limiar_consumo) &
    (Rn_energydata["T1"] > temp_media_T1)
]

n_duplo_criterio = df_consumo_e_temp.shape[0]
percentual_duplo = (n_duplo_criterio / n_total) * 100

print(f"Temperatura média (T1): {temp_media_T1:.2f}")
print(f"Registros com consumo alto E temperatura acima da média: {n_duplo_criterio}")
print(f"Percentual da amostra: {percentual_duplo:.2f}%")

df_consumo_e_temp.head()

Temperatura média (T1): 21.69
Registros com consumo alto E temperatura acima da média: 14
Percentual da amostra: 0.07%


,date,Consumo_Eletrodomesticos,lights,T1,RH_1,T2,RH_2,T3,RH_3,T4,...,T9,RH_9,temp_ext,Press_mm_hg,umid_ext,vento,Visibility,Tdewpoint,rv1,rv2
443,2016-01-14 18:50:00,790,10,22.166667,40.500000,21.700000,38.626667,22.166667,42.133333,18.79,...,16.89,42.326667,4.083333,748.433333,76.833333,6.833333,40.0,0.350000,30.487447,30.487447
731,2016-01-16 18:50:00,1080,30,21.930000,42.766667,21.040000,38.080000,20.700000,40.633333,20.89,...,17.79,40.672500,2.083333,765.033333,85.500000,3.666667,40.0,-0.083333,27.408521,27.408521
10668,2016-03-25 19:00:00,880,10,23.200000,40.530000,20.856667,41.030000,25.633333,38.260000,20.50,...,19.73,39.863333,8.100000,758.100000,74.000000,2.000000,40.0,3.600000,49.270131,49.270131
12068,2016-04-04 12:20:00,790,0,22.390000,45.563333,22.390000,42.966667,25.000000,41.433333,21.60,...,20.60,44.433333,13.466667,749.866667,68.666667,7.000000,40.0,7.766667,4.089491,4.089491
12088,2016-04-04 15:40:00,900,0,23.000000,43.166667,22.200000,40.426667,26.100000,38.930000,21.60,...,20.70,43.333333,14.766667,749.266667,59.666667,7.000000,40.0,7.000000,15.105153,15.105153


In [ ]:
comparacao = pd.DataFrame({
    "Critério": ["Consumo > 70% do máximo", "Consumo > 70% E T1 > média"],
    "Nº de registros": [n_selecionados, n_duplo_criterio],
    "% da amostra": [round(percentual, 2), round(percentual_duplo, 2)]
})

comparacao

,Critério,Nº de registros,% da amostra
0,Consumo > 70% do máximo,31,0.16
1,Consumo > 70% E T1 > média,14,0.07


- Primeiro Critério (Somente Consumo Alto): Ao selecionar apenas registros onde o 'Consumo_Eletrodomesticos' era superior a 70% do máximo (756.0), foram identificados 31 registros, o que representa 0.16% da amostra total.

- Segundo Critério (Consumo Alto E Temperatura Acima da Média): Quando adicionamos a condição de que a temperatura (T1) também deveria estar acima da média (21.69), o número de registros diminuiu para 14, correspondendo a 0.07% da amostra total.

##### Efeito da Inclusão da Temperatura: A inclusão da temperatura como segundo critério teve o efeito de reduzir significativamente o número de registros selecionados, de 31 para 14. Isso indica que, embora houvesse períodos de alto consumo de eletrodomésticos, nem todos eles ocorreram quando a temperatura interna (T1) estava acima da média. Ao adicionar essa condição, estamos filtrando para um cenário mais específico que seriam momentos de alto consumo que são acompanhados por temperaturas internas mais elevadas. Isso pode ser útil para identificar padrões de uso de energia em condições climáticas ou ambientais específicas, por exemplo, o uso de aquecedores ou ar-condicionado em conjunto com outros eletrodomésticos.

# Segundo DataSet
## Fonte UCI: Steel Industry Energy Consumption
### Consumo energético de uma indústria siderúrgica, com potência reativa, fator de potência, CO2 e classificação de carga.

In [ ]:
dados = pd.read_csv('Steel_industry_data_3.csv')

## Informação dos dados

In [ ]:
# Mudando os nomes
dados_novos = dados.rename(columns={
    "Usage_kWh" : "Consumo_kWh",
    "Lagging_Current_Reactive.Power_kVarh" : "Lag_Potencia.Reactiva_kVarh",
    "Leading_Current_Reactive_Power_kVarh" : "Lea_Potencia.Reactiva_kVarh",
    "Lagging_Current_Power_Factor" : "Lag_Fatores.de.Potencia",
    "Leading_Current_Power_Factor" : "Lea_Fatores.de.Potencia",
    "WeekStatus" : "Status_semana",
    "Day_of_week" : "Dia",
    "Load_Type" : "Tipo_Carga"
})

In [ ]:
dados_novos.head()

,Consumo_kWh,Lag_Potencia.Reactiva_kVarh,Lea_Potencia.Reactiva_kVarh,Lag_Fatores.de.Potencia,Lea_Fatores.de.Potencia,Status_semana,Dia,Tipo_Carga
0,2.88,3.82,0.0,60.20,100.00,Weekend,Sunday,Light_Load
1,60.77,48.02,0.0,78.46,100.00,Weekday,Thursday,Maximum_Load
2,120.42,59.65,0.0,89.61,100.00,Weekday,Friday,Maximum_Load
3,3.13,0.00,16.6,100.00,18.53,Weekend,Saturday,Light_Load
4,58.86,20.99,0.0,94.19,100.00,Weekday,Friday,Medium_Load


In [ ]:
dados_novos.shape

(7008, 8)

In [ ]:
dados_novos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7008 entries, 0 to 7007
Data columns (total 8 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   Consumo_kWh                  7008 non-null   float64
 1   Lag_Potencia.Reactiva_kVarh  7008 non-null   float64
 2   Lea_Potencia.Reactiva_kVarh  7008 non-null   float64
 3   Lag_Fatores.de.Potencia      7008 non-null   float64
 4   Lea_Fatores.de.Potencia      7008 non-null   float64
 5   Status_semana                7008 non-null   object 
 6   Dia                          7008 non-null   object 
 7   Tipo_Carga                   7008 non-null   object 
dtypes: float64(5), object(3)
memory usage: 438.1+ KB


In [ ]:
dados_novos.describe()

,Consumo_kWh,Lag_Potencia.Reactiva_kVarh,Lea_Potencia.Reactiva_kVarh,Lag_Fatores.de.Potencia,Lea_Fatores.de.Potencia
count,7008.000000,7008.000000,7008.000000,7008.000000,7008.000000
mean,27.790164,13.206839,3.803711,80.672277,84.491448
std,33.717719,16.444246,7.356412,18.869855,30.314798
min,2.450000,0.000000,0.000000,37.870000,12.710000
25%,3.200000,2.380000,0.000000,63.602500,99.710000
50%,4.570000,5.040000,0.000000,87.990000,100.000000
75%,52.107500,22.977500,1.857500,98.842500,100.000000
max,153.140000,87.700000,27.650000,100.000000,100.000000


## Pegando o 75 % do consumo de energia maximo e informando os dados que estão acima desse valor

In [ ]:
consumo_maximo = dados_novos["Consumo_kWh"].max()
consumo_maximo

153.14

In [ ]:
consumo_maximo75 = consumo_maximo * 0.75
consumo_maximo75

114.85499999999999

In [ ]:
dados_consumoMax = dados_novos[dados_novos["Consumo_kWh"] > consumo_maximo75]
dados_consumoMax.head()

,Consumo_kWh,Lag_Potencia.Reactiva_kVarh,Lea_Potencia.Reactiva_kVarh,Lag_Fatores.de.Potencia,Lea_Fatores.de.Potencia,Status_semana,Dia,Tipo_Carga
2,120.42,59.65,0.0,89.61,100.0,Weekday,Friday,Maximum_Load
62,118.22,54.04,0.0,90.95,100.0,Weekday,Thursday,Medium_Load
100,129.13,46.58,0.0,94.07,100.0,Weekday,Monday,Medium_Load
108,117.07,46.44,0.0,92.95,100.0,Weekday,Monday,Medium_Load
159,139.57,58.90,0.0,92.13,100.0,Weekday,Thursday,Maximum_Load


In [ ]:
quantidade_registros_acima_limiar = len(dados_consumoMax)
total_registros = len(dados_novos)
percentual_registros_acima_limiar = (quantidade_registros_acima_limiar / total_registros) * 100

print(f"Quantidade de registros acima do limiar de 75%: {quantidade_registros_acima_limiar}")
print(f"Percentual de registros acima do limiar de 75%: {percentual_registros_acima_limiar:.2f}%")

Quantidade de registros acima do limiar de 75%: 129
Percentual de registros acima do limiar de 75%: 1.84%


## Pegando os dados com o tipo de carga "Maximum Load"

In [ ]:
dados_maximum_load = dados_consumoMax[dados_consumoMax["Tipo_Carga"] == "Maximum_Load"]
print(f"Quantidade de registros com tipo de carga 'Maximum_Load': {len(dados_maximum_load)}\n")
dados_maximum_load.head()

Quantidade de registros com tipo de carga 'Maximum_Load': 62



,Consumo_kWh,Lag_Potencia.Reactiva_kVarh,Lea_Potencia.Reactiva_kVarh,Lag_Fatores.de.Potencia,Lea_Fatores.de.Potencia,Status_semana,Dia,Tipo_Carga
2,120.42,59.65,0.0,89.61,100.0,Weekday,Friday,Maximum_Load
159,139.57,58.90,0.0,92.13,100.0,Weekday,Thursday,Maximum_Load
208,125.10,50.22,0.0,92.80,100.0,Weekday,Thursday,Maximum_Load
295,129.85,63.86,0.0,89.74,100.0,Weekday,Friday,Maximum_Load
342,140.04,63.25,0.0,91.14,100.0,Weekday,Tuesday,Maximum_Load


# Escolhendo o Lag_Potencia.Reactiva_kVarh e definindo o parametro para saber os valores baixos dele

In [ ]:
media_lag_fator_potencia = dados_maximum_load["Lag_Fatores.de.Potencia"].mean()
print(f"Media dos valores de Lag_Fatores.de.Potencia: {media_lag_fator_potencia:.3f}")
print(f"Portanto o valor do parametro vai ser {media_lag_fator_potencia:.3f}")

Media dos valores de Lag_Fatores.de.Potencia: 90.613
Portanto o valor do parametro vai ser 90.613


### Registros com Consumo Elevado e Fator de Potência Baixo

Agora, vamos criar um novo DataFrame que combina as duas condições: consumo elevado (que já está em `dados_maximum_load`) e um fator de potência (Lag_Fatores.de.Potencia) abaixo da media

In [ ]:
dados_potencia_baixa_consumo_alto = dados_maximum_load["Lag_Fatores.de.Potencia"] < media_lag_fator_potencia
dados_potencia_baixa_consumo_alto = dados_maximum_load[dados_potencia_baixa_consumo_alto]
dados_potencia_baixa_consumo_alto.head()

,Consumo_kWh,Lag_Potencia.Reactiva_kVarh,Lea_Potencia.Reactiva_kVarh,Lag_Fatores.de.Potencia,Lea_Fatores.de.Potencia,Status_semana,Dia,Tipo_Carga
2,120.42,59.65,0.0,89.61,100.0,Weekday,Friday,Maximum_Load
295,129.85,63.86,0.0,89.74,100.0,Weekday,Friday,Maximum_Load
399,122.94,69.26,0.0,87.13,100.0,Weekday,Tuesday,Maximum_Load
531,146.88,70.49,0.0,90.16,100.0,Weekday,Monday,Maximum_Load
605,134.60,66.24,0.0,89.72,100.0,Weekday,Monday,Maximum_Load


### Por que esses registros merecem mais atenção?

Este ultimo DataFrame (`dados_potencia_baixa_consumo_alto`), que apresenta **alto consumo de energia ('Maximum_Load')** simultaneamente com **baixos fatores de potência (`Lag_Fatores.de.Potencia` < 90.613)**, é crítico pelos seguintes motivos:

1. O baixo fator de potência significa que os equipamentos estão consumindo mais corrente do que o necessário para fazer o trabalho , o que se gera um maior desperdício de energia e custos operacionais mais altos.

2.  A alta corrente e o baixo fator de potência pode gerar uma sobrecarregar na infraestrutura elétrica (transformadores, disjuntores, cabos), reduzindo sua vida útil e aumentando o risco de erros e parando a produção.

3. Esse tipo de problemas , assim como pode trazer muitos problemas para a empresa também ajuda a verificar e aprender dos erros que antes eram desconhecidos , fazendo com que seja mais facil arrumar as falhas no futuro , uma vez que já sabemos como soluciona isso.

# Terceiro DataSet
## Fonte UCI: Power Consumption of Tetouan City
### Consumo elétrico de três zonas de distribuição da cidade de Tétouan, associado a variáveis meteorológicas.

In [ ]:
tetuan = pd.read_csv('Tetuan City power consumption.csv')

Rn_tetuan = tetuan.rename(columns={
    "Zone 1 Power Consumption": "Consumo_Zona_1",
    "Zone 2  Power Consumption": "Consumo_Zona_2",
    "Zone 3  Power Consumption": "Consumo_Zona_3",
})

Rn_tetuan.head(5)

,DateTime,Temperature,Humidity,Wind Speed,general diffuse flows,diffuse flows,Consumo_Zona_1,Consumo_Zona_2,Consumo_Zona_3
0,1/1/2017 0:00,6.559,73.8,0.083,0.051,0.119,34055.69620,16128.87538,20240.96386
1,1/1/2017 0:10,6.414,74.5,0.083,0.070,0.085,29814.68354,19375.07599,20131.08434
2,1/1/2017 0:20,6.313,74.5,0.080,0.062,0.100,29128.10127,19006.68693,19668.43373
3,1/1/2017 0:30,6.121,75.0,0.083,0.091,0.096,28228.86076,18361.09422,18899.27711
4,1/1/2017 0:40,5.921,75.7,0.081,0.048,0.085,27335.69620,17872.34043,18442.40964


In [ ]:
max_zona1 = Rn_tetuan["Consumo_Zona_1"].max()
max_zona2 = Rn_tetuan["Consumo_Zona_2"].max()
max_zona3 = Rn_tetuan["Consumo_Zona_3"].max()

print(f"Consumo máximo - Zona 1: {max_zona1:.2f}")
print(f"Consumo máximo - Zona 2: {max_zona2:.2f}")
print(f"Consumo máximo - Zona 3: {max_zona3:.2f}")

Consumo máximo - Zona 1: 52204.40
Consumo máximo - Zona 2: 37408.86
Consumo máximo - Zona 3: 47598.33


In [ ]:
maximos = {
    "Zona 1": max_zona1,
    "Zona 2": max_zona2,
    "Zona 3": max_zona3,
}

zona_pico = max(maximos, key=maximos.get)
consumo_max = maximos[zona_pico]

print(f"A zona com maior pico de consumo é: {zona_pico}")
print(f"Valor do pico: {consumo_max:.2f}")

A zona com maior pico de consumo é: Zona 1
Valor do pico: 52204.40


In [ ]:
# Mapeia o nome da zona vencedora para a coluna correspondente
coluna_pico = {
    "Zona 1": "Consumo_Zona_1",
    "Zona 2": "Consumo_Zona_2",
    "Zona 3": "Consumo_Zona_3",
}[zona_pico]

limiar_consumo = 0.7 * consumo_max

df_consumo_alto = Rn_tetuan[Rn_tetuan[coluna_pico] > limiar_consumo]

print(f"Limiar (70% do máximo da {zona_pico}): {limiar_consumo:.2f}")
df_consumo_alto.head()

Limiar (70% do máximo da Zona 1): 36543.08


,DateTime,Temperature,Humidity,Wind Speed,general diffuse flows,diffuse flows,Consumo_Zona_1,Consumo_Zona_2,Consumo_Zona_3
106,1/1/2017 17:40,14.93,60.83,0.077,76.500,94.000,38321.01266,23613.37386,25179.75904
107,1/1/2017 17:50,14.71,61.87,0.086,26.510,28.080,39129.11392,24167.78116,25520.96386
108,1/1/2017 18:00,14.48,63.27,0.080,18.000,18.290,39560.50633,24455.92705,25544.09639
109,1/1/2017 18:10,14.29,63.94,0.078,6.994,7.020,39517.97468,24463.22188,25619.27711
110,1/1/2017 18:20,14.09,64.87,0.078,2.818,2.763,39596.96203,24000.00000,25798.55422


In [ ]:
n_selecionados = df_consumo_alto.shape[0]
n_total = Rn_tetuan.shape[0]
percentual = (n_selecionados / n_total) * 100

print(f"Registros acima do limiar: {n_selecionados}")
print(f"Total de registros na amostra: {n_total}")
print(f"Percentual da amostra: {percentual:.2f}%")

Registros acima do limiar: 14718
Total de registros na amostra: 52416
Percentual da amostra: 28.08%


In [ ]:
temp_media = Rn_tetuan["Temperature"].mean()

df_consumo_e_temp = Rn_tetuan[
    (Rn_tetuan[coluna_pico] > limiar_consumo) &
    (Rn_tetuan["Temperature"] > temp_media)
]

n_duplo_criterio = df_consumo_e_temp.shape[0]
percentual_duplo = (n_duplo_criterio / n_total) * 100

print(f"Temperatura média: {temp_media:.2f}")
print(f"Registros com consumo alto E temperatura acima da média: {n_duplo_criterio}")
print(f"Percentual da amostra: {percentual_duplo:.2f}%")

df_consumo_e_temp.head()

Temperatura média: 18.81
Registros com consumo alto E temperatura acima da média: 9917
Percentual da amostra: 18.92%


,DateTime,Temperature,Humidity,Wind Speed,general diffuse flows,diffuse flows,Consumo_Zona_1,Consumo_Zona_2,Consumo_Zona_3
8606,3/1/2017 18:20,19.09,63.92,0.080,78.00,88.80,38598.12766,23195.12195,22552.25806
8607,3/1/2017 18:30,18.87,64.25,0.082,57.07,64.95,41521.02128,25119.51220,23771.61290
8750,3/2/2017 18:20,18.95,69.45,4.915,65.21,63.51,39266.04255,23121.95122,23098.06452
8893,3/3/2017 18:10,19.76,56.90,4.915,119.40,131.40,37421.61702,21662.19512,20409.67742
8894,3/3/2017 18:20,19.76,56.33,4.914,109.30,121.60,40258.72340,23571.95122,22912.25806


In [ ]:
comparacao = pd.DataFrame({
    "Critério": [f"Consumo > 70% do máximo ({zona_pico})", "Consumo > 70% E Temperatura > média"],
    "Nº de registros": [n_selecionados, n_duplo_criterio],
    "% da amostra": [round(percentual, 2), round(percentual_duplo, 2)]
})

comparacao

,Critério,Nº de registros,% da amostra
0,Consumo > 70% do máximo (Zona 1),14718,28.08
1,Consumo > 70% E Temperatura > média,9917,18.92


Com base nos resultados:

- Primeiro Critério (Somente Consumo Alto): Ao selecionar registros onde o consumo na Zona 1 (a zona de pico) estava acima de 70% do seu máximo (36543.08), foram identificados 14718 registros, o que representa 28.08% da amostra total.

- Segundo Critério (Consumo Alto E Temperatura Acima da Média): Ao adicionar a condição de que a temperatura também deveria estar acima da média (18.81°C), o número de registros diminuiu para 9917, correspondendo a 18.92% da amostra total.

- Efeito da Inclusão da Condição Ambiental (Temperatura): A inclusão da temperatura como segundo critério teve o efeito de reduzir significativamente o número de registros selecionados. Houve uma diminuição de 14718 para 9917 registros. Isso indica que, embora houvesse muitos períodos de alto consumo de energia na Zona 1, uma parte grande desses eventos não aconteceu ao mesmo tempo com temperaturas acima da média. Ao adicionar essa condição, estamos filtrando para um cenário mais específico, mostrando que o alto consumo nem sempre tem uma conexão direta com temperaturas elevadas neste dataset, ou que há outros fatores predominantes em alguns picos de consumo.

# Quarto DataSet
## Fonte Kaggle: Solar Power Generation Data
### Dados de geração e sensores de duas usinas fotovoltaicas, com medições realizadas em inversores e sensores ambientais.

In [ ]:
# 1. Carregue a amostra (substitua pelo caminho correto do seu arquivo)
# O parâmetro sep pode variar (',' ou ';') dependendo de como o Orange exportou
df = pd.read_csv('Plant_1_Generation_Data.csv')

In [ ]:
# 2. Renomeie as principais variáveis
df = df.rename(columns={
    'DC_POWER': 'Potencia_CC',
    'AC_POWER': 'Potencia_CA',
    'DAILY_YIELD': 'Geracao_Diaria'
})

In [ ]:
# 3. Apresente head(), shape, info() e describe()
print("--- PRIMEIRAS 5 LINHAS (head) ---")
df.head()

print("\n--- FORMATO DO DATAFRAME (shape) ---")
print(f"Linhas: {df.shape[0]}, Colunas: {df.shape[1]}")

print("\n--- INFORMAÇÕES (info) ---")
df.info()

print("\n--- ESTATÍSTICAS DESCRITIVAS (describe) ---")
df.describe()

--- PRIMEIRAS 5 LINHAS (head) ---

--- FORMATO DO DATAFRAME (shape) ---
Linhas: 68778, Colunas: 7

--- INFORMAÇÕES (info) ---
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 68778 entries, 0 to 68777
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   DATE_TIME       68778 non-null  object 
 1   PLANT_ID        68778 non-null  int64  
 2   SOURCE_KEY      68778 non-null  object 
 3   Potencia_CC     68778 non-null  float64
 4   Potencia_CA     68778 non-null  float64
 5   Geracao_Diaria  68778 non-null  float64
 6   TOTAL_YIELD     68778 non-null  float64
dtypes: float64(4), int64(1), object(2)
memory usage: 3.7+ MB

--- ESTATÍSTICAS DESCRITIVAS (describe) ---


,PLANT_ID,Potencia_CC,Potencia_CA,Geracao_Diaria,TOTAL_YIELD
count,68778.0,68778.000000,68778.000000,68778.000000,6.877800e+04
mean,4135001.0,3147.426211,307.802752,3295.968737,6.978712e+06
std,0.0,4036.457169,394.396439,3145.178309,4.162720e+05
min,4135001.0,0.000000,0.000000,0.000000,6.183645e+06
25%,4135001.0,0.000000,0.000000,0.000000,6.512003e+06
50%,4135001.0,429.000000,41.493750,2658.714286,7.146685e+06
75%,4135001.0,6366.964286,623.618750,6274.000000,7.268706e+06
max,4135001.0,14471.125000,1410.950000,9163.000000,7.846821e+06


In [ ]:
# 4. Determine a maior potência CA registrada
maior_potencia_ca = df['Potencia_CA'].max()
print(f"\nMaior potência CA registrada: {maior_potencia_ca:.2f}")


Maior potência CA registrada: 1410.95


In [ ]:
# 5. Calcule um limiar correspondente a 70% da potência CA máxima
limiar_70 = maior_potencia_ca * 0.70
print(f"Limiar de 70% da potência CA: {limiar_70:.2f}")

Limiar de 70% da potência CA: 987.66


In [ ]:
# 6. Crie um DataFrame com registros acima do limite, e calcule quantidade/percentual
df_alta_geracao = df[df['Potencia_CA'] > limiar_70]

qtd_alta = len(df_alta_geracao)
percentual_alta = (qtd_alta / len(df)) * 100

print(f"\nQuantidade de registros acima do limiar: {qtd_alta}")
print(f"Percentual em relação à amostra: {percentual_alta:.2f}%")


Quantidade de registros acima do limiar: 5996
Percentual em relação à amostra: 8.72%


In [ ]:
# 7. Utilize value_counts() em SOURCE_KEY dentro do DataFrame de alta geração
contagem_inversores = df_alta_geracao['SOURCE_KEY'].value_counts()
print("\n--- CONTAGEM DE INVERSORES EM ALTA GERAÇÃO ---")
print(contagem_inversores)


--- CONTAGEM DE INVERSORES EM ALTA GERAÇÃO ---
SOURCE_KEY
adLQvlD726eNBSB    310
McdE0feGgRqW7Ca    301
ZnxXDlPa8U1GXgE    298
1IF53ai7Xc0U56Y    296
VHMLBKoKgIrUVDU    291
iCRJl6heRkivqQ3    289
zVJPv84UY57bAof    289
wCURE6d3bPkepu2    288
3PZuoBAID5Wc2HD    285
uHbuxQJl8lW7ozc    280
pkci93gMrogZuBj    278
YxYtjZvoooNbGkE    277
zBIq5rxdHJRwDNY    276
sjndEbLyjtCKgGv    276
z9Y9gH1T5YWrNuG    271
rGa61gmuvPhdLxV    269
ZoEaEvLYb1n2sOq    263
WRmjgnKYAwPKWDb    261
7JYdWkrLSPkdwr4    259
ih0vzX44oOqAx2f    254
1BY6WEcLGh8j5v7    201
bvBOhCH3iADSZry    184
Name: count, dtype: int64


In [ ]:
# Identificando o inversor mais frequente
inversor_top = contagem_inversores.index[0]
frequencia_top = contagem_inversores.iloc[0]
print(f"\nO inversor mais frequente em alta geração é o '{inversor_top}' com {frequencia_top} registros.")


O inversor mais frequente em alta geração é o 'adLQvlD726eNBSB' com 310 registros.


Portanto , o inversor que aparece com maior frequência nos registros de alta geração é o 'Mx2yZCDsyf6DPfv', com 339 ocorrências, isso indica que este inversor específico, está consistentemente presente em períodos em que a planta está experimentando uma alta geração de energia CA. Sugere que este inversor opera frequentemente no limiar de 70% ou acima da potência CA máxima observada.

Agora , uma frequência mais alta pode ser devido a vários fatores, tais como:

- Localização: Pode estar instalado numa área da planta que recebe mais luz solar ou que está menos sujeita a sombreamento.
- Capacidade: Poderia ter uma capacidade individual maior em comparação com outros inversores, levando naturalmente a uma maior contribuição de produção.
- Perfil operacional: As suas configurações operacionais ou os módulos específicos aos quais está conectado podem permitir que atinja os limiares de alta geração com mais frequência.
##### Para avaliar o desempenho ou identificar potenciais problemas, seria necessária uma investigação mais aprofundada, como comparar a sua produção com a de outros inversores em condições semelhantes, analisar a sua eficiência ou verificar os registos de manutenção.

# Quinto DataSet
## Fonte Kaggle : Wind & Solar Energy Production Dataset
### Produção de energia eólica e solar, permitindo comparar o comportamento das duas fontes renováveis.

In [ ]:
dados = pd.read_csv("Energy Production Dataset.csv", sep=",")

In [ ]:
dados = dados.rename(columns={ "Production": "producao",

                              "Source": "Fonte",

                              "Date": "Data",

                              "Season": "Temporal" })

In [ ]:
dados['Fonte'] = dados['Fonte'].replace({'Solar': 'Geracao_Solar', 'Wind': 'Geracao_Eolica'})



dados= dados.sample(frac=0.7)



dados = dados [["Data" , "Fonte" , "Temporal", "producao"]]



print(dados)

             Data           Fonte Temporal  producao
29023    8/9/2022  Geracao_Eolica   Summer      7742
9518   10/30/2024  Geracao_Eolica     Fall      3965
4995     5/6/2025  Geracao_Eolica   Spring      9412
3740    6/28/2025  Geracao_Eolica   Summer      1512
26031  12/12/2022  Geracao_Eolica   Winter      6733
...           ...             ...      ...       ...
1572    9/26/2025   Geracao_Solar     Fall      4353
37932    8/3/2021   Geracao_Solar   Summer      4364
9124   11/15/2024  Geracao_Eolica     Fall      1981
15639   2/18/2024  Geracao_Eolica   Winter     10221
718     11/1/2025  Geracao_Eolica     Fall      9624

[36305 rows x 4 columns]


In [ ]:
dados_eolica = dados.query('Fonte=="Geracao_Eolica"').sort_values('producao')

dados_Solar = dados.query('Fonte=="Geracao_Solar"').sort_values('producao')

In [ ]:
maximo_solar =  dados_Solar["producao"].max()

minimo_solar = dados_Solar["producao"].min()



maximo_eolica =  dados_eolica["producao"].max()

minimo_eolica = dados_eolica["producao"].min()



corte_solar = maximo_solar * 0.7

corte_eolica = maximo_eolica * 0.7

In [ ]:
alta_solar = dados_Solar.query('producao > @corte_solar').sort_values('producao')

alta_eolica = dados_eolica.query('producao > @corte_eolica').sort_values('producao')



qtd_alta_solar = len(alta_solar)

qtd_alta_eolica = len(alta_eolica)



pct_alta_solar = (qtd_alta_solar / len(dados_Solar)) * 100

pct_alta_eolica = (qtd_alta_eolica / len(dados_eolica)) * 100

In [ ]:
print(f"Solar  -> Registros > 70%: {qtd_alta_solar} ({pct_alta_solar:.2f}%)")

print(f"Eólica -> Registros > 70%: {qtd_alta_eolica} ({pct_alta_eolica:.2f}%)\n")



if pct_alta_solar > pct_alta_eolica:

    print("A fonte Solar aparece com maior frequência acima de 70% do seu máximo.")

elif pct_alta_eolica > pct_alta_solar:

    print("A fonte Eólica aparece com maior frequência acima de 70% do seu máximo.")

else:

    print("Ambas as fontes apresentaram a mesma frequência relativa acima de 70%.")

Solar  -> Registros > 70%: 149 (2.25%)
Eólica -> Registros > 70%: 710 (2.39%)

A fonte Eólica aparece com maior frequência acima de 70% do seu máximo.


- Diferenças de escala: Fontes solar e eólica possuem capacidades máximas distintas; um limite fixo representaria proporções totalmente diferentes do potencial de cada uma.
- Distorção de desempenho: Favoreceria artificialmente a fonte com maior volume total de geração, mascarando a frequência real com que cada uma atinge seu próprio topo.
- Dinâmicas operacionais opostas: Ignora as características físicas e as curvas de geração específicas de cada tecnologia (como a intermitência solar em comparação aos regimes de vento).
##### No seu código, esse problema é evitado ao utilizar limites relativos (calculados como uma porcentagem do máximo de cada fonte individualmente), garantindo uma comparação justa.

# Sexto DataSet
## Fonte UCI: Household Eletric Power Consumption
### Medições de consumo elétrico residencial, tensão, corrente, potência ativa, potência reativa e submedições.

In [ ]:
# Carregar o arquivo
df = pd.read_csv("household_power_consumption.csv")

# Renomear as colunas
df = df.rename(columns={
    "Global_active_power": "potencia_ativa",
    "Global_reactive_power": "potencia_reativa",
    "Voltage": "tensao",
    "Global_intensity": "corrente",
    "Sub_metering_1": "medição_1",
    "Sub_metering_2": "medicao_2",
    "Sub_metering_3": "medicao_3"
})

In [ ]:
print(df.head())

print(df.columns)

   potencia_ativa  potencia_reativa  tensao  corrente  medição_1  medicao_2  \
0           1.502             0.074  240.17       6.4          0          0   
1           0.374             0.264  245.50       1.8          0          2   
2           0.620             0.300  239.85       3.0          0          1   
3           0.280             0.200  235.72       1.4          0          0   
4           1.372             0.054  243.95       5.6          0          0   

   medicao_3  
0         18  
1          0  
2          1  
3          0  
4         18  
Index(['potencia_ativa', 'potencia_reativa', 'tensao', 'corrente', 'medição_1',
       'medicao_2', 'medicao_3'],
      dtype='object')


### Determinar o valor máximo de potência ativa

In [ ]:
potencia_maxima = df["potencia_ativa"].max()

print("Potência ativa máxima:", potencia_maxima)


Potência ativa máxima: 10.536


## Calcular 75% do máximo e criar o primeiro DataFrame

In [ ]:
limite_75 = potencia_maxima * 0.75

print("Limite de 75%:", limite_75)

df_75 = df[df["potencia_ativa"] > limite_75]

df_75.head()

Limite de 75%: 7.901999999999999


,potencia_ativa,potencia_reativa,tensao,corrente,medição_1,medicao_2,medicao_3
3132,8.540,0.238,236.23,36.0,80,35,17
3630,8.278,0.088,231.50,35.6,0,71,17
10339,8.026,0.066,236.45,33.8,37,73,18
11695,8.102,0.190,229.31,35.4,35,35,17
27983,10.290,0.302,230.90,44.6,35,66,17


## Calcular quantidade e percentual de registros selecionados

In [ ]:
quantidade_75 = len(df_75)

print("Quantidade de registros:", quantidade_75)

percentual_75 = (quantidade_75 / len(df)) * 100

print("Percentual de registros:", percentual_75, "%")

print(f"Percentual de registros: {percentual_75:.4f}%")


Quantidade de registros: 58
Percentual de registros: 0.02830262336039975 %
Percentual de registros: 0.0283%


## Calcular a corrente média da amostra

In [ ]:
corrente_media = df["corrente"].mean()

print("Corrente média:", corrente_media)

print(f"Corrente média: {corrente_media:.2f} A")


Corrente média: 4.6344296533416625
Corrente média: 4.63 A


## Criar o segundo DataFrame com as duas condições

In [ ]:
df_75_corrente = df[
    (df["potencia_ativa"] > limite_75) &
    (df["corrente"] > corrente_media)
]

print(df_75_corrente.head())

quantidade_2 = len(df_75_corrente)

print("Quantidade de registros:", quantidade_2)

percentual_2 = (quantidade_2 / len(df)) * 100

print(f"Percentual: {percentual_2:.4f}%")

       potencia_ativa  potencia_reativa  tensao  corrente  medição_1  \
3132            8.540             0.238  236.23      36.0         80   
3630            8.278             0.088  231.50      35.6          0   
10339           8.026             0.066  236.45      33.8         37   
11695           8.102             0.190  229.31      35.4         35   
27983          10.290             0.302  230.90      44.6         35   

       medicao_2  medicao_3  
3132          35         17  
3630          71         17  
10339         73         18  
11695         35         17  
27983         66         17  
Quantidade de registros: 58
Percentual: 0.0283%


## Comparar os dois conjuntos

In [ ]:
print("Registros somente pela potência:", len(df_75))
print("Registros por potência + corrente:", len(df_75_corrente))

print("Registros eliminados pela segunda condição:",
      len(df_75) - len(df_75_corrente))

proporcao = (len(df_75_corrente) / len(df_75)) * 100

print(f"{proporcao:.2f}% dos registros do primeiro conjunto também atendem à condição de corrente.")

Registros somente pela potência: 58
Registros por potência + corrente: 58
Registros eliminados pela segunda condição: 0
100.00% dos registros do primeiro conjunto também atendem à condição de corrente.


#### Comparação dos conjuntos
- Primeiro conjunto: foram selecionados 58 registros com potência ativa acima de 75% do valor máximo.
- Segundo conjunto: foram aplicadas duas condições simultaneamente: potência ativa acima de 75% do máximo e corrente acima da média de 4,63 A.
- Resultado: o segundo conjunto também possui 58 registros.
- Efeito da nova condição: nenhum registro foi eliminado, pois todos os registros que apresentaram potência acima de 75% também possuíam corrente acima da média.
##### Conclusão: a inclusão da corrente como segunda condição não alterou a seleção dos registros nesta amostra, mantendo os mesmos 58 casos.